packages/grid/src/vaadin-grid-scroll-mixin.js

```diff
@@ -123,6 +123,17 @@ export const ScrollMixin = (superClass) =>
       this.$.items.addEventListener('focusin', (e) => {
         const itemsIndex = e.composedPath().indexOf(this.$.items);
         this._rowWithFocusedElement = e.composedPath()[itemsIndex - 1];
+
+        // Make sure the row with the focused element is fully inside the visible viewport
+        this.__scrollIntoViewport(this._rowWithFocusedElement.index);
+
+        if (this._rowWithFocusedElement && !this.$.table.contains(e.relatedTarget)) {
```

Comment:
Here,  you're checking if the `_rowWithFocusedElement` is defined, but on the line above, you're accessing the `index` property of the same object. Is this check needed? Or should the call in line 128 have a check if the object is defined as well?

Reply:
True, wrapped the referenced line inside the check also